### PR TITLE
feat(limits): apply limits to cpu for pythonrunner and deployments if…

### PR DESCRIPTION
**Pull request by HknLof**
… provided

### CODE REVIEW

File: platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java [hunk 1]
```diff
@@ -48,4 +51,42 @@ private DataCaterK8sConfig() {}
...
+  static final String PYTHONRUNNER_RESOURCES_REQUESTS_MEMORY =
+      ConfigProvider.getConfig()
+          .getOptionalValue("datacater.pythonrunner.resources.requests.memory", String.class)
+          .orElse("200Mi");
+  static final Optional<String> PYTHONRUNNER_RESOURCES_LIMITS_CPU =
+      ConfigProvider.getConfig()
+          .getOptionalValue("datacater.pythonrunner.resources.limits.cpu", String.class);
+  static final String PYTHONRUNNER_RESOURCES_LIMITS_MEMORY =
+      ConfigProvider.getConfig()
+          .getOptionalValue("datacater.pythonrunner.resources.requests.memory", String.class)
```
Comment (flippingbits) [inline platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java:70]: Should be `datacater.pythonrunner.resources.limits.memory`.

File: platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java [hunk 2]
```diff
@@ -48,4 +51,42 @@ private DataCaterK8sConfig() {}
...
+    Map<String, Quantity> requests =
+        Map.of(
+            MEMORY,
+            new Quantity(PYTHONRUNNER_RESOURCES_REQUESTS_MEMORY),
+            CPU,
+            new Quantity(PYTHONRUNNER_RESOURCES_REQUESTS_CPU));
+    Map<String, Quantity> limits =
+        PYTHONRUNNER_RESOURCES_LIMITS_CPU
+            .map(
+                s ->
```
Comment (flippingbits) [inline platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java:83]: Can you please rename `s` to something meaningful?

File: platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java [hunk 3]
```diff
@@ -48,4 +51,42 @@ private DataCaterK8sConfig() {}
...
+      ConfigProvider.getConfig()
+          .getOptionalValue("datacater.pythonrunner.resources.requests.memory", String.class)
+          .orElse("200Mi");
+  static final Optional<String> PYTHONRUNNER_RESOURCES_LIMITS_CPU =
+      ConfigProvider.getConfig()
+          .getOptionalValue("datacater.pythonrunner.resources.limits.cpu", String.class);
+  static final String PYTHONRUNNER_RESOURCES_LIMITS_MEMORY =
+      ConfigProvider.getConfig()
+          .getOptionalValue("datacater.pythonrunner.resources.requests.memory", String.class)
+          .orElse("0.5");
```
Comment (flippingbits) [inline platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java:71]: `0.5` does not make sense as a default value for the memory limit. This sounds more like it was meant for CPU.

File: platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java [hunk 4]
```diff
@@ -48,4 +51,42 @@ private DataCaterK8sConfig() {}
...
+  private static final String CPU = "cpu";
+  private static final String MEMORY = "memory";
+  static final String PYTHONRUNNER_RESOURCES_REQUESTS_CPU =
+      ConfigProvider.getConfig()
+          .getOptionalValue("datacater.pythonrunner.resources.requests.cpu", String.class)
+          .orElse("0.1");
+  static final String PYTHONRUNNER_RESOURCES_REQUESTS_MEMORY =
+      ConfigProvider.getConfig()
+          .getOptionalValue("datacater.pythonrunner.resources.requests.memory", String.class)
+          .orElse("200Mi");
```
Comment (flippingbits) [inline platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java:64]: `200Mi` feels too much. In my Minikube cluster, I observed a memory usage of ~ `95Mi` when not using any requests/limits.

**Review by flippingbits — CHANGES_REQUESTED** [review]
Thanks for this PR! :) I left a few comments.

File: platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java (hunk 3 above)
Comment (flippingbits) [reply platform-api/src/main/java/io/datacater/core/kubernetes/DataCaterK8sConfig.java:71]: Might just use the same default as for the request to follow the best practice of setting memory request=memory limit?

**Comment by HknLof** [comment]
@flippingbits Could you check again, I updated the blocks accordingly. Introduced `100Mi` for memory limit and request.

**Review by flippingbits — APPROVED** [review]
LGTM, thanks for the iteration

**Pull request merged**